whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotGenerationWindow.java

```diff
@@ -0,0 +1,53 @@
+package net.ripe.db.nrtm4;
+
+import net.ripe.db.nrtm4.domain.NrtmVersionInfo;
+import net.ripe.db.whois.common.DateTimeProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+@Service
+public class SnapshotGenerationWindow {
```

Comment:
Window doesn't have to be configurable, we can hard-code a daily time to create a snapshot

Reply:
the RFC says: "3.2.2 The mirror server MUST generate a new Snapshot File between once per hour and once per day, if there have been changes to the IRR objects." so the idea of this 'window' is to prevent snapshots less than 1 hour old from being overwritten, in the case that the NRTM db is reset and the server rebooted just before midnight (say) and the snapshot timer ticks at midnight.

Reply:
Can you run a scheduled job more frequently then and generate a snapshot if an existing one is >24h old? Then there's no need for a window.

Reply:
not really, because then the time since the last snapshot will be 24h 14m since it takes 14m to generate (at least, it does on dev & prepdev) and we won't be compliant with the RFC. if we assume that it takes 14m and look for an expiry date at at 23h 47m, we must assume that 14m is consistent across all production machines every day, which is unlikely to be the case. so then the exact time of generation will slide every day, forwards or backwards depending on which machine picks up the job and how busy it happens to be. some days we might not generate a file in 14m and then we'll be non-compliant again. using the window gives us the best chance of remaining compliant while not ever needing manual intervention to reset the time of snapshots if the time slips too much in one direction. it's also an easy change to define two windows in one day, for example, to make very sure we'll always be compliant and hit the 'at least one snapshot every 24h' requirement. if you still want to generate a snapshot at a fixed time then of course that's fine, but i think it's important to understand why i wrote the code this way, and the implications of the fixed time / fixed expiry approaches.